inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/ConceptFeatureSupport.java

```diff
@@ -159,9 +156,15 @@ public String renderFeatureValue(AnnotationFeature aFeature, AnnotationFS aFs,
     @Override
     public void setFeatureValue(JCas aJcas, AnnotationFeature aFeature, int aAddress, Object aValue)
     {
-        KBHandle kbInst = (KBHandle) aValue;
         FeatureStructure fs = selectByAddr(aJcas, FeatureStructure.class, aAddress);
-        setFeature(fs, aFeature, kbInst != null ? kbInst.getIdentifier() : null);
+        String kbIdentifier;
+        if (aValue instanceof KBHandle) {
+            kbIdentifier = ((KBHandle) aValue).getIdentifier();
+        }
+        else {
+            kbIdentifier = (String) aValue;
```

Comment:
Better create a KBHandle for the identifier before calling this method and remove the instanceof here. Method should always be called with a KBHandle.